### DISCUSSION TITLE
Scanning Assemblies using AttributesModel in .net core produces ReflectionTypeLoadException

### DISCUSSION THREAD

**Discussion by chkp-omeran** [dadhi/DryIoc#528] [General]
I noticed the following behavior in .net 6 (which as I understand is the same in .net Core) - 
If I have an assembly a.dll which depends on assembly b.dll, but my main application only references a.dll, using AttributesModel.Scan on a.dll will result in ReflectionTypeLoadException, because unlike .net Framework, .net core doesn't attempt to load indirect dependencies when loading types.
I can probably get around it by reading the ".deps.json" file of each DLL and looking for dependent assemblies, but maybe there's a simpler way to do it?

**Comment by dadhi** [comment]
@chkp-omeran Hi, that's the interesting problem. I don't have a simple way from the top of my head. 
But I would've looked at how https://github.com/khellang/Scrutor does these things.
